data/src/main/java/me/androidbox/data/worker_manager/UploadEventWorker.kt

```diff
@@ -0,0 +1,100 @@
+package me.androidbox.data.worker_manager
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.Data
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import me.androidbox.data.R
+import me.androidbox.data.remote.model.response.EventDto
+import me.androidbox.data.remote.model.response.PhotoDto
+import me.androidbox.data.remote.network.event.EventService
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.data.worker_manager.UploadEventImp.Companion.ERROR
+import me.androidbox.data.worker_manager.UploadEventImp.Companion.EVENT_PHOTOS
+import me.androidbox.data.worker_manager.util.CreatePhotoMultipart
+import okhttp3.MultipartBody
+
+@HiltWorker
+class UploadEventWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val eventService: EventService,
+    private val createPhotoMultipart: CreatePhotoMultipart,
+    private val moshi: Moshi
+) : CoroutineWorker(context, workerParameters)  {
+
+    override suspend fun doWork(): Result {
+        val listOfPhoto = inputData.getStringArray(UploadEventImp.EVENT_PHOTOS)?.toList()
+        val eventRequestJson = inputData.getString(UploadEventImp.EVENT)
+        val isEditMode = inputData.getBoolean(UploadEventImp.IS_EDIT_MODE, false)
+
+        val formData = getFormData(isEditMode)
+
+        if(!eventRequestJson.isNullOrEmpty()) {
+            val listOfPhotoMultiPart = listOfPhoto?.let {
+                /* Create the multipart for the list of photos */
+                createPhotoMultipart.createMultipartPhotos(listOfPhoto)
+            } ?: listOf()
+
+            val responseResult = checkResult {
+                eventService.createEvent(
+                    listOfPhoto = listOfPhotoMultiPart,
+                    eventBody = MultipartBody.Part.createFormData(formData, eventRequestJson)
+                )
+            }
+
+            val result = responseResult.fold(
+                onSuccess = { eventDto ->
+                    if(eventDto.photos.isNotEmpty()) {
+                        val outputData = photoToJson(eventDto)
+
+                        Result.success(outputData)
+                    }
```

Comment:
@philipplackner @MKiperszmid Will return the photos url and keys in the Success, if there any photos that was uploaded when creating or updating an event. Is this correct?

Reply:
Yea but the question is, do you really care about the response of the Success?

Reply:
@MKiperszmid I am sure that @philipplackner said we get back the key and url in the response. And these are the keys we use when we want to delete the photos from the BE. Maybe @philipplackner can confirm what we need to do with this response.

Reply:
Yes correct @steve1rm . You need these keys and the URLs are also only known after the API responds